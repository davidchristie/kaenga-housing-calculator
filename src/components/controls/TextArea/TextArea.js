import PropTypes from 'prop-types'
import React, { Component } from 'react'
import {
  ControlLabel,
  FormControl,
  FormGroup,
  HelpBlock
} from 'react-bootstrap'

class TextArea extends Component {
  render () {
    const {
      input: { onChange, value },
      label,
      meta: { error },
      placeholder,
      rows,
      style
    } = this.props
    return (
      <FormGroup>
        <ControlLabel>{label}</ControlLabel>
        <FormControl
          componentClass='textarea'
          onChange={onChange}
          placeholder={placeholder}
          rows={rows}
          value={value}
          style={style}
        />
        {error && <HelpBlock>{error}</HelpBlock>}
      </FormGroup>
    )
  }
}

TextArea.defaultProps = {
  input: {},
  meta: {}
}

TextArea.propTypes = {
  input: PropTypes.shape({
    onChange: PropTypes.func,
    value: PropTypes.any
  }).isRequired,
  label: PropTypes.string,
  placeholder: PropTypes.string
}

export default TextArea
