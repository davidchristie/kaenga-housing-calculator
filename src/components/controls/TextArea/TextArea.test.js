import React from 'react'
import ReactTestUtils from 'react-dom/test-utils'

import TextArea from './TextArea'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<TextArea />)).toMatchSnapshot()
})
