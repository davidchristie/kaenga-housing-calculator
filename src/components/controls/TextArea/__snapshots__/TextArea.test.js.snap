exports[`test matches snapshot 1`] = `
<FormGroup
  bsClass="form-group">
  <ControlLabel
    bsClass="control-label"
    srOnly={false} />
  <FormControl
    bsClass="form-control"
    componentClass="textarea"
    onChange={undefined}
    placeholder={undefined}
    rows={undefined}
    style={undefined}
    value={undefined} />
</FormGroup>
`;
